### ISSUE TITLE
Fulltext indexes are created even when full text search is disabled

### ISSUE THREAD

**Issue by sesom42** [friendica/friendica#2690]
The full-text indexes are always created, no matter how the switch is set in the settings. For large installations that can quickly consume a lot of useless disk space. If there is a switch that affects the database model, the database must also be changed dynamically.

Currently the tables "profile" and "item" uses many full-text indexes. Almost a third of the space is once again consumed with the indices:

```
# du -ch item* profile* | tail -1
1,1G    total
# du -ch FTS* | tail -1
403M    total
```

Is the switch in the settings really needed?

**Comment by annando** [comment]
The switch in the settings is for the case that you use an SQL server that isn't capable of full text  indexes (older InnoDB versions).

I really would like to remove the full text search in Friendica because it only works on really small installations and I haven't find a way to speed it up in a reliable way.

**Comment by rabuzarus** [comment]
Oh now on my system it works fine

**Comment by sesom42** [comment]
If the full-text index is supported only by very few installations then it should be removed.

An alternative approach might be to parse the subject and text of a posting, to remove stop-words (if, when, etc.) and then fill a separate table with these words and references to the original posting. This could be more efficient because the database does not know anything about stop-words.

**Comment by annando** [comment]
The full text index is supported by most Friendica installations since most are still running with MyISAM.

Concerning the stop words: The database *does* know about them: https://dev.mysql.com/doc/refman/5.6/en/fulltext-stopwords.html

Only thing is that we would need to define the additional ones.

**Comment by sesom42** [comment]
Nice to know. Thx. Your link includes information to implenent an own stopword-table. But in shared hosting environment almost no users has the right "super" to use "set global". Then an own full-text stop word table is not possible.

**Comment by Hypolite** [comment]
Closed via #3354
